crates/webidl/src/first_pass.rs

```diff
@@ -1,3 +1,12 @@
+//! Because some WebIDL constructs are defined in multiple places
+//! (keyword `partial` is used to add to an existing construct),
+//! We need to first walk the webidl to collect all non-partial
+//! constructs so that we have containers in which to put the
+//! partial ones.
+//!
+//! Only `interface`s, `dictionary`s, `enum`s and `mixin`s can
+//! be partial.
+
```

Comment:
A couple of more things that have been on my mind and should go in the first pass in the future (likely not exhaustive):
- `typedef`s should not be exposed on the rust side, but should be processed by the binding's generator (i.e. this crate). To do that, `typedef` declarations have to be recorded before building the AST, so in the first pass.
- should we want to implement up-casting conversion by more than one level, we need to be able to walk the inheritance tree up to the root while building the AST, doing so would require recording that in the first pass

More generally, the fist pass is used to record information that needs to be globally accessible while building the backend AST.